src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/service/impl/MessageRejectionServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package uk.gov.companieshouse.chipsrestinterfacesconsumer.service.impl;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.kafka.support.serializer.DeserializationException;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.common.ApplicationLogger;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.service.MessageRejectionService;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.slack.SlackMessagingService;
+
+@Service
+public class MessageRejectionServiceImpl implements MessageRejectionService {
+
+    @Autowired
+    private ApplicationLogger logger;
+
+    @Autowired
+    private SlackMessagingService slackMessagingService;
+
+    @Override
+    public void handleRejectedMessage(Exception exception, ConsumerRecord<?, ?> datum) {
+        String errorMessageDetails = buildMessage(datum.topic(), datum.partition(), datum.offset());
+        if (exception.getCause() instanceof DeserializationException) {
+             sendToSlack(exception, errorMessageDetails);
+        } else {
+            logger.error(String.format("Kafka consumer message exception - %s", errorMessageDetails), exception);
+        }
+    }
+
+    private void sendToSlack(Exception exception, String errorMessageDetails) {
+        String deserializationFailureMessage = String.format("Failed to deserialize message - %s", errorMessageDetails);
+        logger.error(deserializationFailureMessage, exception);
```

Comment:
this log statement should be in the handleRejectedMessage method, not here in the sendToSlack method.
perhaps move this to line 24?

Reply:
It should log just the same here but if you perfer I'll move it.

Reply:
my thought process being by logging it in the "parent" method, we dont need to pass the exception to the sendToSlack method.
And secondly, the logging of the exception and sending to slack are two seperate things.
calling a method called sendToSlack I wouldn't expect it to log out a deserialization error message